featuretools/primitives/standard/transform_primitive.py

```diff
@@ -623,10 +631,108 @@ class NumericLag(TransformPrimitive):
     def __init__(self, periods=1, fill_value=None):
         self.periods = periods
         self.fill_value = fill_value
+        warnings.warn(
+            "NumericLag is deprecated and will be removed in a future version. Please use the 'Lag' primitive instead.",
+            FutureWarning,
+        )
 
     def get_function(self):
         def lag(time_index, numeric):
             x = pd.Series(numeric.values, index=time_index.values)
             return x.shift(periods=self.periods, fill_value=self.fill_value).values
 
         return lag
+
+
+class Lag(TransformPrimitive):
+    """Shifts an array of values by a specified number of periods.
+
+    Args:
+        periods (int): The number of periods by which to shift the input.
+            Default is 1. Periods correspond to rows.
+
+        fill_value (int, float, str, bool, optional): The value to use to fill in
+            the gaps left after shifting the input. Default is None.
+
+    Examples:
+        >>> lag = Lag()
+        >>> lag([1, 2, 3, 4, 5], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
+        [nan, 1.0, 2.0, 3.0, 4.0]
+
+        You can specify the number of periods to shift the values
+
+        >>> lag_periods = Lag(periods=3)
+        >>> lag_periods(["hello", "world", "test", "foo", "bar"], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
+        [nan, nan, nan, 'hello', 'world']
+
+        You can specify the fill value to use
+
+        >>> lag_fill_value = Lag(fill_value=100)
+        >>> lag_fill_value([1, 2, 3, 4], pd.Series(pd.date_range(start="2020-01-01", periods=4, freq='D'))).tolist()
+        [100, 1, 2, 3]
+    """
+
+    name = "lag"
+    input_types = [
```

Comment:
@ozzieD I forgot about return types when suggesting you expand the specified logical types. I think nate's probably right that we can keep this simple.

Reply:
@thehomebrewnerd So I confirmed Integer and Boolean inputs do work because `init_ww_and_concat_fm` will replace the output type with the nullable counterparts. @rwedge mentioned this would be a problem with stacked primitives.

Reply:
@rwedge Can you provide a little more detail on the stacking problem and is it something we need to be concerned with in this PR?

Reply:
@thehomebrewnerd with `return_type = None` features created by this primitive will use the first input's column schema as their return type.  When returning the final feature matrix, `init_ww_and_concat_fm` will convert some columns with non-nullable return types to nullable types if there are nulls present.  However, this does not happen inside `FeatureSetCalculator` after values are calculated, so if a primitive unable to handle nulls stacked on this primitive it could receive null values as input.

Reply:
@rwedge I see. Is that enough of an edge case for now that we can wait to address, or do you think that is something we should resolve now?

Reply:
@thehomebrewnerd a sanity check would be to see if there are any agg primitives in our primitive libraries that would encounter this scenario

Reply:
Changed inputs to 
- ColumnSchema(semantic_tags={"category"})
- ColumnSchema(semantic_tags={"numeric"})
- ColumnSchema(logical_type=BooleanNullable)
- ColumnSchema(logical_type=Boolean)

Tried using a bunch of aggs with Lag and haven't seen any issues